data/code/lib.c

```diff
@@ -589,3 +593,738 @@ void check_lap_ex()
 #endif
 }
 #endif
+
+#define EXTENDED_TERRAIN_BOUNCE_FLAG 0x1
+#define EXTENDED_TERRAIN_BOUNCE_LIFTOFF_FLAG 0x2
+
+#define CRS_GROUND_COLLISION_TRIANGLE_POINTER_OFFSET 0x20
+#define COLLISION_TRIANGLE_TERRAIN_TYPE_OFFSET 0x16
+#define COLLISION_TRIANGLE_SPLASH_HASH_OFFSET 0x20
+#define EXTENDED_TERRAIN_BOUNCY 0xB0
+
+// The importance of this remaining in its own section is that its functionality will be needed for
+// any future custom terrain types added, within the Extended Terrain Types patch or otherwise.
+#if EXTENDED_TERRAIN_TYPES
+
+typedef struct
+{
+    int point_indexes[3];
+    float unknown;
+    short normal[3];
+    char terrain_type;
+    char terrain_type_part_two;
+    // ...
+} CollisionTriangle;
+
+// Check against all of the custom material flags enabled by the patch.
+int is_extended_terrain_type(const char terrain_type)
+{
+    return terrain_type == EXTENDED_TERRAIN_BOUNCY;
+}
+
+int should_return_fake_code(char* const ground)
+{
+    const int* const triangle_ptr =
+        (int*)(ground + COLLISION_TRIANGLE_SPLASH_HASH_OFFSET);  // Pointer to the triangle's
```

Comment:
What's the difference between `CRS_GROUND_COLLISION_TRIANGLE_POINTER_OFFSET` and `COLLISION_TRIANGLE_SPLASH_HASH_OFFSET`? For some reason, like in the `get_splash_code_inline()` below, I thought `CRS_GROUND_COLLISION_TRIANGLE_POINTER_OFFSET` would be used here.

My preference is still to declare the `CrsGround` structure, and use the type in all signatures, alleviating the cognitive complexity in all places at once:

```cpp
bool should_return_fake_code(const CrsGround* const ground)
{
    const CollisionTriangle* const triangle = ground->triangle;
    if (triangle)
    {
        return is_extended_terrain_type(triangle->terrain_type);
    }
    return 0;
}
```